### ISSUE TITLE
[Bug]: Wrong Background Image URL

### ISSUE THREAD

**Issue by ManOki** [nextcloud/server#34842]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

I've updated from NC24 to NC25. Now, the app-background.jpg is not loaded, resulting in white icons on white background.

### Steps to reproduce

1. Login into Nextcloud
2. Background Image URL is https://server.hosted.org/core/img/app-background.jpg without /nextcloud/ and therefore not loaded.

### Expected behavior

The Background Image URL should respect config.php parameter "overwritewebroot": "/nextcloud" 
and thus be https://server.hosted.org/nextcloud/core/img/app-background.jpg
If I enter this URL into my webbrowser, the image is shown.

### Installation method

Community Docker image

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.1

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwriteprotocol": "https",
        "overwritehost": "server.hosted.org",
        "overwritewebroot": "\/nextcloud",
        "overwrite.cli.url": "https:\/\/server.hosted.org\/nextcloud",
        "htaccess.RewriteBase": "\/",
        "version": "25.0.0.18",
        "updatechecker": false,
        "installed": true,
        "forcessl": true,
        "trusted_domains": [
            "server.hosted.org"
        ],
        "loglevel": 3,
        "logtimezone": "UTC",
        "default_phone_region": "DE",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trashbin_retention_obligation": "auto",
        "dbtype": "mysql",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 6379
        },
        "filelocking.enabled": true,
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "enable_previews": true,
        "preview_max_filesize_image": -1,
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "maintenance": false,
        "theme": "",
        "defaultapp": "apporder",
        "updater.release.channel": "stable",
        "auth.bruteforce.protection.enabled": false
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 2.17.0
  - admin_audit: 1.15.0
  - calendar: 4.0.1
  - cloud_federation_api: 1.8.0
  - contacts: 5.0.1
  - contactsinteraction: 1.6.0
  - dav: 1.24.0
  - federatedfilesharing: 1.15.0
  - files: 1.20.1
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.13.0
  - mail: 2.0.3
  - notifications: 2.13.1
  - oauth2: 1.13.0
  - photos: 2.0.0
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.2
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - text: 3.6.0
  - twofactor_backupcodes: 1.14.0
  - viewer: 1.9.0
  - workflowengine: 2.7.0
Disabled:
  - apporder: 0.15.0
  - bruteforcesettings: 1.2.0
  - circles: 0.17.8
  - comments: 1.5.0
  - dashboard: 7.0.0
  - encryption
  - federation: 1.5.0
  - firstrunwizard: 1.1
  - impersonate: 1.11.0
  - nextcloud_announcements: 1.3.0
  - password_policy: 1.5.0
  - sharebymail: 1.5.0
  - support: 1.0.0
  - survey_client: 1.0.0
  - suspicious_login
  - systemtags: 1.2.0
  - theming: 1.6.0
  - twofactor_totp
  - updatenotification: 1.5.0
  - user_ldap
  - user_status: 1.0.0
  - weather_status: 1.0.0
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Additional info

Setup: 
Docker Image https://github.com/nextcloud/docker/blob/master/.examples/dockerfiles/full/apache/Dockerfile
Reverse Proxy https://docs.nextcloud.com/server/25/admin_manual/configuration_server/reverse_proxy_configuration.html

**Comment by 4ch1m** [comment]
I can confirm this bug.

It has severe side-effects in my setup.
I use _fail2ban_ together with my webserver.
All requests outside my white-listed context roots will trigger a complete ban of the requester's IP.

The `default.css` is the culprit:
https://github.com/nextcloud/server/blob/d007088cf5d89e29065991e0cbe2c890dfa13d96/apps/theming/css/default.css#L57

This URL is absolute and ignores the (configurable) root-/context-path.

### NOTE:

The **LOGIN** page will fetch this CSS:

```
.../apps/theming/css/default.css?v=...
```

(Which contains the bug.)

Whereas **after** the login this CSS is being fetched:

```
.../apps/theming/theme/default.css?plain=...
```

(In which the correct root-path is available!)

**Comment by ManOki** [comment]
Temporary Fix: `sed -i "s=url('/=url('/nextcloud/=g" apps/theming/css/default.css`

**Comment by PVince81** [comment]
@Pytal can you verify ?

**Comment by Pytal** [comment]
Thanks for reporting this @ManOki

For Nextcloud 25 we have moved much of our logic into the theming app which handles URL path generation

The paths mentioned above in the thread are set as fallbacks and options such as `overwritewebroot` are taken into account within the theming app

This has been fixed in 25.0.1, and you may refer to our wiki page for release date information https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule

**Comment by ManOki** [comment]
Confirm fixed in 25.0.1

**Comment by 4ch1m** [comment]
I can also confirm that this is fixed with the latest docker-container `25.0.1`.
Thanks!
